feat(curriculum): forum leaderboard project (#50241)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-fetch-promises-and-async-await-by-building-an-fcc-forum-leaderboard/643c8d9a73965c0a26d7f634.md

---
id: 643c8d9a73965c0a26d7f634
title: Step 28
challengeType: 0
dashedName: step-28
---

# --description--

The `topic_list` object contains 4 more properties, but you only need the `topics` array. In this step, you will destructure the `topics` array from the `topic_list` object.

# --hints--

> You should use a `const` variable to destructure the `topics` variable from the `topic_list` object.

Reviewer: ```suggestion
You should use `const` to destructure the `topics` variable from the `topic_list` object.
```

Author: updated

---
id: 643c8d9a73965c0a26d7f634
title: Step 28
challengeType: 0
dashedName: step-28
---

# --description--

The `topic_list` object contains 4 more properties, but you only need the `topics` array. In this step, you will destructure the `topics` array from the `topic_list` object like this 

```js

> const  { topics } = topic_list;

Reviewer 2: I think we should remove the example here because campers will have had a lot practice with destructuring in earlier projects.

Author: done

---
id: 643c8d9a73965c0a26d7f634
title: Step 11
challengeType: 0
dashedName: step-11
---

# --description--

> The `topic_list` object contains 4 more properties, but you only need the `topics` array. In this step, you will destructure the `topics` array from the `topic_list` object.

Reviewer 3: ```suggestion
The `topic_list` object contains several properties, but you only need the `topics` array.

Destructure the `topics` array from the `topic_list` object.
```

Destructure the `topics` array from the `topic_list` object.

# --hints--

You should use `const` to destructure the `topics` variable from the `topic_list` object.

```js

> assert(code.match(/{\s+const\s+{topic_list, users}\s+=\s+data;\s+const\s+{\s*topics\s*}\s+=\s+topic_list/g));

Reviewer 2: I am unable to pass this test with the correct code

<img width="865" alt="Screenshot 2023-11-16 at 12 41 43 PM" src="https://github.com/freeCodeCamp/freeCodeCamp/assets/67210629/681c8708-30c6-42ac-93ea-f586af4e4573">

Author: done

}

.support {
  background-color: var(--light-orange);
}

.general {
  background-color: var(--light-green);
}

.javascript {
  background-color: var(--golden-yellow);
}

.backend {
  background-color: var(--gold);
}

.python {
  background-color: var(--green);
}

.motivation {
  background-color: var(--peach);
}

.avatar-container {
  display: flex;
  justify-content: center;
  gap: 10px;
  flex-wrap: wrap;
}

.avatar-container img {
  width: 30px;
  height: 30px;
}

@media (max-width: 750px) {
  .table-wrapper {
    padding: 0 15px;
  }

  table {
    width: 700px;
  }

  th {
    font-size: 1.2rem;
  }

  .post-title {
    font-size: 1.1rem;
  }
}
```

```js
const FORUM_LATEST = "https://forum-proxy.freecodecamp.rocks/latest";
const FORUM_TOPIC_URL = "https://forum.freecodecamp.org/t/";
const FORUM_CATEGORY_URL = "https://forum.freecodecamp.org/c/";
const AVATAR_URL = "https://sea1.discourse-cdn.com/freecodecamp";

const fetchData = async () => { 
  try{
    const res = await fetch(FORUM_LATEST);
    const data = await res.json();
    console.log(data)
    showLatestPosts(data);
  }
  catch(err){
    console.log(err)
  }
}

fetchData();

--fcc-editable-region--

> const showLatestPosts = (data) => {
>   const {topic_list, users} = data;
>   
> }

Reviewer 3: ```suggestion
const showLatestPosts = (data) => {
  const { topic_list, users } = data;
  
};
```
